src/cogent3/core/new_alphabet.py

```diff
@@ -771,17 +779,17 @@ def to_index(self, seq) -> numpy.ndarray:
         raise TypeError(f"{type(seq)} not supported")
 
     @to_index.register
-    def _(self, seq: str) -> numpy.ndarray:
+    def _(self, seq: str) -> int:
```

Comment:
is seq is a single character? if yes should we change the arg name to char

Reply:
On CharAlphabet yes, but on KmerAlphabet and CodonAlphabet it will be > 1 character.